src-tauri/src/structs.rs

```diff
@@ -1,11 +1,23 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
 use citadel_internal_service_connector::connector::WrappedSink;
 use citadel_internal_service_connector::io_interface::tcp::TcpIOInterface;
 use citadel_internal_service_types::InternalServiceResponse;
 use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc::Sender;
 use tokio::sync::Mutex;
+use uuid::Uuid;
+
+pub struct PacketHandle {
+    pub request_id: Uuid, // The ID to listen for in the response stream
+    pub channel: Sender<InternalServiceResponse>, // The channel to stream the response to
+}
```

Comment:
Your intuition for handling sending and receiving is generally correct. I basically do the same thing in this PR: https://github.com/Avarok-Cybersecurity/citadel-internal-service/pull/40